index.js

```diff
@@ -23,6 +25,22 @@ const BOOL_TRUE = Buffer.from([1])
 const BOOL_FALSE = Buffer.from([0])
 
 const encoder = {
+  ssbURI(input) {
+    let [type, format, data = ''] = new URL(input).pathname.split('/')
+
+    type = NAMED_TYPES[type]
+    format = type.formats[format]
```

Comment:
What if `type` is undefined here? E.g. if the `input` is `ssb:address/multiserver/whatever`, which is a valid SSB URI. The `type` would be `address` and it's not defined in NAMED_TYPES.

Reply:
Then an error will be thrown. Ebay would you like the error to say?